mphys/scenario_aerothermal.py

```diff
@@ -0,0 +1,70 @@
+import openmdao.api as om
+from .scenario import Scenario
+from .coupling_aerothermal import CouplingAeroThermal
+
+
+class ScenarioAeroThermal(Scenario):
+
+    def initialize(self):
+        """
+        A class to perform a single discipline aerodynamic case.
+        The Scenario will add the aerodynamic builder's precoupling subsystem,
+        the coupling subsystem, and the postcoupling subsystem.
+        """
+        self.options.declare('aero_builder', recordable=False,
+                              desc='The Mphys builder for the aerodynamic solver')
+        self.options.declare('thermal_builder', recordable=False,
+                              desc='The Mphys builder for the Thermal solver')
+        self.options.declare('thermalxfer_builder', recordable=False,
+                              desc='The Mphys builder for the heat flux and temperature transfer')
+        self.options.declare('in_MultipointParallel', default=False,
+                             desc='Set to `True` if adding this scenario inside a MultipointParallel Group.')
+        self.options.declare('geometry_builder', default=None, recordable=False,
+                             desc='The optional Mphys builder for the geometry')
+
+    def setup(self):
+        aero_builder = self.options['aero_builder']
+        thermal_builder = self.options['thermal_builder']
+        thermalxfer_builder = self.options['thermalxfer_builder']
+        geometry_builder = self.options['geometry_builder']
+
+        if self.options['in_MultipointParallel']:
+            self._mphys_initialize_builders(aero_builder, thermal_builder,
+                                            thermalxfer_builder, geometry_builder)
+            self._mphys_add_mesh_and_geometry_subsystems(aero_builder, thermal_builder,
+                                                         geometry_builder)
+
+        self.mphys_add_pre_coupling_subsystem('aero', aero_builder, self.name)
```

Comment:
It seems that this function name is deprecated, and now it is called `_mphys_add_pre_coupling_subsystem_from_builder` in scenario.py. Please double-check.

Reply:
Yes, Ping is correct. This change was to allow user-defined subsystems to be added to scenarios in the base class' `setup` method.